pkg/event/processor.go

```diff
@@ -44,6 +44,7 @@ type BatchEventProcessor struct {
 	Mux             sync.Mutex
 	Ticker          *time.Ticker
 	EventDispatcher Dispatcher
+	processing      bool
```

Comment:
This is like a poor man's Mutex :)

Reply:
you noticed that.  Yeah, I set it and test it in the same thread and then set it back in the other thread.  So, there is really no chance of collision.  However, go lint disagreed.  So, I went ahead and created a new util struct called AtomicProperty.